### PR TITLE
feat: Add MUI Config and Theme Provider

### PR DESCRIPTION
**Pull request by mertcanaltin**
Add MUI Config and Theme Provider

issue: #5306

### CODE REVIEW

**Comment by AugustinMauroy** [comment]
Hum what are you doing ? If I have good understanding of the issue we should add default style for icon. But here it's not that.

Can you give me explaining?

**Comment by mertcanaltin** [comment]
> Provider

hi I tried to add MUI Provider but I guess I failed :(

**Comment by AugustinMauroy** [comment]
I thinks we should do an thinks like [that](https://github.com/react-icons/react-icons#configuration)

File: pages/_app.mdx [hunk 1]
```diff
@@ -3,25 +3,33 @@ import { NodeDataProvider } from '../providers/nodeDataProvider';
...
 import { SiteProvider } from '../providers/siteProvider';
 import sourceSansPro from '../util/sourceSansPro';
+import { ThemeProvider, createTheme } from '@mui/material/styles';
 
 import '../styles/old/index.scss';
 
+const theme = createTheme({
+  typography: {
+    fontFamily: [
+      'Source Sans Pro',
```
Comment (ovflowd) [inline pages/_app.mdx]: You should probably use the sourceSansPro variable that is imported as it gives you the correct font family name 👀

Comment (ovflowd) [reply pages/_app.mdx]: Like replace the first line with sourceSansPro.style.fontFamily

**Review by ovflowd — COMMENTED** [review]
Looking good! I left a comment just because I believe fonts would be broken without that 👀

**Comment by ovflowd** [comment]
@mertcanaltin looks like you forgot to commit the updates on the package-lock. Can you run npm i and commit the package-lock changes? 🙃

**Review by ovflowd — APPROVED** [review]
LGTM! But build is failing, please check for lining issues :) + You forgot to add the package-lock file, again.

**Comment by mertcanaltin** [comment]
> LGTM! But build is failing, please check for lining issues :) + You forgot to add the package-lock file, again.

i think i am getting an acorn error 🤔

**Comment by ovflowd** [comment]
An acorn error? @nodejs/website can somebody look into this and update this PR with the changes? (package-lock)

File: package.json [hunk 2]
```diff
@@ -36,7 +36,10 @@
     "storybook:build": "cross-env NODE_ENV=production NODE_NO_WARNINGS=1 storybook build"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.0",
+    "@emotion/styled": "^11.11.0",
```
Comment (Harkunwar) [inline package.json:40]: These are `peer-dependencies` that `MUI` depends on

File: pages/_app.mdx [hunk 3]
```diff
@@ -3,28 +3,37 @@ import { NodeDataProvider } from '../providers/nodeDataProvider';
 import { LocaleProvider } from '../providers/localeProvider';
 import { SiteProvider } from '../providers/siteProvider';
 import sourceSansPro from '../util/sourceSansPro';
+import { ThemeProvider, createTheme } from '@mui/material/styles';
 
 import '../styles/old/index.scss';
 
+export const theme = createTheme({
```
Comment (Harkunwar) [inline pages/_app.mdx:10]: Needed to add `export` otherwise it isn't treated as TSX but MD instead.

**Review by Harkunwar — APPROVED** [review]
LGTM!

File: package.json (hunk 2 above)
Comment (ovflowd) [reply package.json:40]: Don't we use emotion already? 🤔

**Review by ovflowd — APPROVED** [review]
LGTM!

**Pull request merged**